cluster-agent/controllers/argoproj.io/namespace_reconciler.go

```diff
@@ -72,7 +72,8 @@ func (r *ApplicationReconciler) startTimerForNextCycle(ctx context.Context, name
 			cleanOrphanedCRsfromCluster_Operation(ctx, r.DB, r.Client, log)
 
 			// Recreate Secrets that are required by Applications, but missing from cluster.
-			recreateClusterSecrets(ctx, r.DB, r.Client, log)
+			recreateClusterSecrets_ManagedEnvironments(ctx, r.DB, r.Client, log)
+			recreateClusterSecrets_RepositoryCredentials(ctx, r.DB, r.Client, log)
```

Comment:
The new function has some code in common with the original recreateClusterSecrets_ManagedEnvironments function. I wonder if it can be commoned up?  If not simple to refactor, then leave it.

Reply:
It is updated, PTAL.